test/components/previews/modal_component_preview.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class ModalComponentPreview < ViewComponent::Preview
+  def default
+    render Viral::ModalComponent.new(title: 'This is a modal title') do |modal|
+      modal.body do
```

Comment:
UPDATE: I've reached out about this request. I purposely did not include a button/link in the component. because I wanted the modal to function like antd. There may be times where we might not need a link/button. For example, announcements within IRIDA.

Reply:
Please see latest changes.

Reply:
Decided to remove this functionality for now.